front/components/workspace/ProviderManagement.tsx

```diff
@@ -0,0 +1,127 @@
+import { ContextItem, Modal, SliderToggle } from "@dust-tt/sparkle";
+import type { ModelProviderIdType, WorkspaceType } from "@dust-tt/types";
+import { SUPPORTED_MODEL_CONFIGS } from "@dust-tt/types";
+import { MODEL_PROVIDER_IDS } from "@dust-tt/types";
+import { useState } from "react";
+
+import { MODEL_PROVIDER_LOGOS } from "@app/components/assistant_builder/InstructionScreen";
+
+interface ModelSelectionModalProps {
+  owner: WorkspaceType;
+  showProviderModal: boolean;
+  onClose: () => void;
+  onSave: (updateProviders: ModelProviderIdType[]) => void;
+}
+
+type ProviderStates = Record<ModelProviderIdType, boolean>;
+
+const prettyfiedProviderNames: { [key in ModelProviderIdType]: string } = {
+  openai: "Open AI",
+  anthropic: "Anthropic",
+  mistral: "Mistral AI",
+  google_ai_studio: "Google",
+};
+
+// TODO: Jules 06/06/2024: use selection modal in workspace/index.tsx once Model Deactivation ready
+export function ProviderSelectionModal({
+  owner,
+  showProviderModal,
+  onClose,
+  onSave,
+}: ModelSelectionModalProps) {
+  const enabledProviders: ModelProviderIdType[] =
+    owner.whiteListedProviders ?? [...MODEL_PROVIDER_IDS];
+  const initialProviderStates: ProviderStates = enabledProviders.reduce(
+    (acc, provider) => {
+      acc[provider] = true;
+      return acc;
+    },
+    {} as ProviderStates
+  );
```

Comment:
Could be wrap in a `useMemo()` to only recompute when `owner.whiteListedProviders` changes.